### PR TITLE
Minzoom

### PR DESCRIPTION
**Pull request by Pomax**
fixes https://github.com/mozilla/webmaker-android/issues/2170 by capping the allowed scale for an element - the longest edge on the element is not allowed to be below 44 pixels as indicated by the `minScaledEdgeLength` value.

This seemed a good candidate for statics'ing, so it's simply a utility function on `BasicElement` instead of being an instance function.

### CODE REVIEW

File: www_src/components/basic-element/basic-element.jsx [hunk 1]
```diff
@@ -11,13 +11,29 @@ var touchhandler = require("../../lib/touchhandler");
...
       link: require('./types/link.jsx')
-    }
+    },
+    /**
+     * A static function for determining whether a given scale will
+     * be safe, or needs capping because it would lead to an element
+     * that is visually too small to manipulate.
+     */
+    safifyScale: (function() {
+      var minScaledEdgeLength = 44;
```
Comment (k88hudson) [inline www_src/components/basic-element/basic-element.jsx]: make this a static

File: www_src/components/basic-element/basic-element.jsx [hunk 2]
```diff
@@ -11,13 +11,29 @@ var touchhandler = require("../../lib/touchhandler");
...
+      return function(component, scale) {
+        var node  = component.getDOMNode(),
+            style = getComputedStyle(node,null),
+            w = parseInt(style.getPropertyValue("width"), 10),
```
Comment (k88hudson) [inline www_src/components/basic-element/basic-element.jsx]: single quotes

File: www_src/components/basic-element/basic-element.jsx [hunk 3]
```diff
@@ -11,13 +11,29 @@ var touchhandler = require("../../lib/touchhandler");
...
+            h = parseInt(style.getPropertyValue("height"), 10),
+            e = w>h ? w : h;
+        if(e*scale < minScaledEdgeLength) { scale = minScaledEdgeLength/e; }
+        return scale;
+      };
+    }())
```
Comment (k88hudson) [inline www_src/components/basic-element/basic-element.jsx]: why is this wrapped in an anonymous function?

Comment (Pomax) [inline www_src/components/basic-element/basic-element.jsx]: closure around minScaledEdgeLength  - if we use a static, that is no longer necessary

File: www_src/components/basic-element/basic-element.jsx (hunk 1 above)
Comment (Pomax) [inline www_src/components/basic-element/basic-element.jsx]: :+1:

**Comment by k88hudson** [comment]
A few tiny comments. Works great! R+

**Pull request merged**